core/src/test/scala/org/apache/spark/scheduler/ReplayListenerSuite.scala

```diff
@@ -218,9 +218,7 @@ class ReplayListenerSuite extends SparkFunSuite with BeforeAndAfter with LocalSp
 
     // Verify the same events are replayed in the same order
     assert(sc.eventLogger.isDefined)
-    val originalEvents = sc.eventLogger.get.loggedEvents.filter { e =>
-      !JsonProtocol.sparkEventFromJson(e).isInstanceOf[SparkListenerStageExecutorMetrics]
-    }
+    val originalEvents = sc.eventLogger.get.loggedEvents
```

Comment:
@edwinalu can you remind us what the point of this old filter was?

I know it had something to do with not logging every event, only the aggregated ones, to keep event logs small, but forget the details.  @wypoon can you explain why it needed to change?

Reply:
My change simply reverts originalEvents to what it had been before @edwinalu changed it. In ReplayListenerSuite, "End-to-end replay" and "End-to-end replay with compression" were failing with the filter, and removing the filter made them pass. It seems that SparkListenerStageExecutorMetrics events now show up in replayedEvents with my changes.

Reply:
Sorry for the delay. I also don't remember the details. It is aggregating the metrics, and logging only at stage end, but this should be true for both, and removing the filter seems right, looking at the code now.